pkg/compiler/compilation_document.go

```diff
@@ -53,6 +61,22 @@ func (doc *CompilationDocument) OutputTo(dest string) error {
 			}
 			_, err = f.WriteTo(file)
 			file.Close()
+
+			fileExt := filepath.Ext(path)
+			if strings.HasPrefix(fileExt, ".") {
```

Comment:
nit: `fileExt` will always have a "." or be empty

Reply:
As in, I could have just done `if fileExt != ""`? If so, yeah — but I find the fact that it starts with `"."` non-obvious (I have to re-look-it-up each time), so if I went down that path I would have gone with:

```go
if fileExt != "" {
	fileExt = fileExt[1:] // if Ext returns non-empty, it's always prefixed by "."
	...
```

and by then it feels a bit potato-potato to me.

Reply:
I was thinking more along the lines of:
```go
if fileExt != "" {
    fileExt = strings.TrimPrefix(fileExt, ".")
```
note that TrimPrefix will no-op if the prefix isn't found.

either way, just an "nit" so feel free to disregard